app/objects/c_operation.py

```diff
@@ -220,24 +220,18 @@ async def _execute_atomically(self, services):
         cannot be executed. Will do a loop once through all abilities
         enumerated in adversary.
         """
+        itera = iter(range(0, len(self.adversary.atomic_ordering)))
```

Comment:
defaults to 0. Also `itera` is kinda non-descript... ability_iter maybe?

Reply:
`ability_iter = iter(range(len(self.adversary.atomic_ordering))`

Reply:
Makes sense.